packages/server/src/core/internals/procedureBuilder.ts

```diff
@@ -249,6 +251,18 @@ export function createBuilder<TConfig extends AnyRootConfig>(
       return createNewBuilder(_def, builder._def) as any;
     },
     use(middlewareBuilderOrFn) {
+      if (
+        typeof middlewareBuilderOrFn === 'object' &&
```

Comment:
Crazy checks just because `function` was already taken as a type, but `.use` seems the most sensible home for this.